x/service/keeper/msg_server_add_service.go

```diff
@@ -38,15 +38,22 @@ func (k msgServer) AddService(
 	}
 
 	// Check if the service already exists or not.
-	if _, found := k.GetService(ctx, msg.Service.Id); found {
-		logger.Error(fmt.Sprintf("Service already exists: %v", msg.Service))
+	foundService, found := k.GetService(ctx, msg.Service.Id)
+	if found {
+		if foundService.OwnerAddress != msg.Service.OwnerAddress {
+			logger.Error(fmt.Sprintf("Owner address of existing service (%q) does not match the owner address %q", foundService.OwnerAddress, msg.Address))
+			return nil, types.ErrServiceInvalidOwnerAddress.Wrapf(
+				"existing owner address %q does not match the new owner address %q",
+				foundService.OwnerAddress, msg.Service.OwnerAddress,
+			)
+		}
 		return nil, types.ErrServiceAlreadyExists.Wrapf(
-			"duplicate service ID: %s", msg.Service.Id,
+			"TODO: Don't have for updating existing service ID: %s", msg.Service.Id,
 		)
 	}
 
-	// Retrieve the address of the actor adding the service.
-	accAddr, err := sdk.AccAddressFromBech32(msg.Address)
+	// Retrieve the address of the actor adding the service; the owner of the service.
+	serviceOwnerAddr, err := sdk.AccAddressFromBech32(msg.Address)
```

Comment:
Updating `proto/poktroll/shared/tx.proto#MsgAddService` with:
```
option (cosmos.msg.v1.signer) = "owner_address";
string owner_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```
Would allow us to have `msg.OwnerAddress`

Reply:
Great suggestion!